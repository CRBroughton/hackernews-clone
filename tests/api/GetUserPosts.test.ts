// tests/Post.test.ts
import { createTestContext } from '../__helpers'
const ctx = createTestContext()

it('Gets posts from the a specific user', async() => {
  const feed = await ctx.client.request(`
      query {
        getUserPosts(id: "97527cbe-2de0-40cd-a953-8caae780e66e") {
          id
          description
          url
          voters {
            id
            name
            email
          }
        } 
      }
    `)
  expect(feed).toMatchSnapshot()
  const persistedData = await ctx.prisma.post.findMany(
    { where: { postedById: '97527cbe-2de0-40cd-a953-8caae780e66e' } },
  )
  expect(persistedData).toMatchSnapshot()
})
