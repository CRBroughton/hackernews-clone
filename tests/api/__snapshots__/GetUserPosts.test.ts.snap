// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gets posts from the a specific user 1`] = `
Object {
  "getUserPosts": Array [
    Object {
      "description": "Code-First GraphQL schemas for JavaScript/TypeScript",
      "id": "16e11628-462a-4235-afb2-21c0db7425ba",
      "url": "https://nexusjs.org",
      "voters": Array [],
    },
    Object {
      "description": "My Portfolio",
      "id": "2df7837b-788d-4410-aa65-919da54f7175",
      "url": "https://crbroughton.me",
      "voters": Array [],
    },
    Object {
      "description": "Vite.js",
      "id": "040688e1-c08f-4752-8d5f-888ffb36ef02",
      "url": "https://vitejs.dev",
      "voters": Array [],
    },
  ],
}
`;

exports[`Gets posts from the a specific user 2`] = `
Array [
  Object {
    "createdAt": 2022-02-15T20:24:21.706Z,
    "description": "Code-First GraphQL schemas for JavaScript/TypeScript",
    "id": "16e11628-462a-4235-afb2-21c0db7425ba",
    "postedById": "97527cbe-2de0-40cd-a953-8caae780e66e",
    "topic": "General",
    "url": "https://nexusjs.org",
  },
  Object {
    "createdAt": 2022-02-15T20:24:21.706Z,
    "description": "My Portfolio",
    "id": "2df7837b-788d-4410-aa65-919da54f7175",
    "postedById": "97527cbe-2de0-40cd-a953-8caae780e66e",
    "topic": "General",
    "url": "https://crbroughton.me",
  },
  Object {
    "createdAt": 2022-02-15T20:24:21.706Z,
    "description": "Vite.js",
    "id": "040688e1-c08f-4752-8d5f-888ffb36ef02",
    "postedById": "97527cbe-2de0-40cd-a953-8caae780e66e",
    "topic": "General",
    "url": "https://vitejs.dev",
  },
]
`;
